modules/optimization/src/reporters/ParsedScalarReporter.C

```diff
@@ -0,0 +1,54 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ParsedScalarReporter.h"
+
+registerMooseObject("OptimizationApp", ParsedScalarReporter);
+
+InputParameters
+ParsedScalarReporter::validParams()
+{
+  InputParameters params = ParsedReporterBase::validParams();
+  params.addRequiredParam<std::vector<ReporterName>>("reporter_names", "Reporter names ");
+  params.addClassDescription("Apply parsed functions to scalar entries held in reporters.");
+  return params;
+}
+
+ParsedScalarReporter::ParsedScalarReporter(const InputParameters & parameters)
+  : ParsedReporterBase(parameters),
+    _output_reporter(
+        declareValueByName<Real>(getParam<std::string>("name"), REPORTER_MODE_REPLICATED))
+{
+  // get reporters to operate on
+  const std::vector<ReporterName> reporter_names(
+      getParam<std::vector<ReporterName>>("reporter_names"));
+  if (reporter_names.size() != _reporter_symbols.size())
+    mooseError(
```

Comment:
Should be a param error on one of the parameters and avoid using newlines in parameter errors

Reply:
The new lines make it easier to read.  Do you really want them gone?